src/processes/topic-auto-msg-process.js

```diff
@@ -0,0 +1,75 @@
+const {Op} = require('sequelize');
+const {UserTopic} = require('../databases/models');
+const db = require('../databases/models');
+
+const StreamChat = require('stream-chat').StreamChat;
+require('dotenv').config();
+
+const topicAutoMessageProcess = async () => {
+  const serverClient = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    let comunity_message_format = await db.sequelize.query(
+      `SELECT
+        (created_at + interval '1d' * delay_time)::date as current_date,
+        now()::date,
+        community_message_format_id, user_id, topic_id, message, delay_time, created_at AS createdAt, updated_at AS updatedAt
+      FROM community_message_format AS CommunityMessageFormat
+      WHERE CommunityMessageFormat.status = '1'
+      AND (created_at + interval '1d' * delay_time)::date = now()::date`
+    );
+    comunity_message_format = comunity_message_format[0];
+
+    const topic_ids = comunity_message_format.map((item) => item.topic_id);
+
+    const user_topics = await UserTopic.findAll({
+      where: {
+        topic_id: {
+          [Op.in]: topic_ids
+        },
+        notify_user: true,
+        is_anonymous: false
+      }
+    });
+
+    await Promise.all([
+      user_topics.forEach(async (user_topic) => {
+        const message_format = await comunity_message_format.filter(
+          (message_format) => message_format.topic_id === user_topic.topic_id
+        );
+        const senderUserId = message_format[0].user_id;
+        const targetUserId = user_topic.user_id;
+
+        let chat = serverClient.channel('messaging', {
+          type_channel: 0,
+          members: [senderUserId, targetUserId],
+          created_by_id: senderUserId
+        });
+        await chat.create();
+
+        const channelState = await chat.watch();
+
+        if (channelState.messages.length === 0) {
+          await chat.sendMessage({
+            text: message_format[0].message,
+            user_id: targetUserId
+          });
+
+          try {
+            await chat.stopWatching();
+          } catch (error) {
+            console.log('::: Error on stopWatching :::', error);
+          }
+        }
+      })
+    ]);
+
+    return comunity_message_format;
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `topicAutoMessageProcess` is doing too many things: querying the database, processing data, creating chat channels, sending messages, etc. This violates the Single Responsibility Principle (SRP) of SOLID principles. Consider breaking down this function into smaller ones, each with a single responsibility.

Also, error handling could be improved. Currently, if an error occurs during the execution of the function, it will stop the entire process. Instead, consider handling errors at a more granular level, so that if one operation fails, it doesn't prevent others from executing.

For example, when iterating over `user_topics` and sending messages, if an error occurs for one user, it shouldn't prevent messages from being sent to other users. You can achieve this by adding a try-catch block inside the forEach loop.

Lastly, avoid using console.log for error logging in production code. Consider using a proper logging library that provides more control over log levels, formats, destinations, etc.

```diff
-         try {
-           await chat.stopWatching();
-         } catch (error) {
-           console.log('::: Error on stopWatching :::', error);
-         }
+         try {
+           await chat.stopWatching();
+         } catch (error) {
+           logger.error('::: Error on stopWatching :::', error);
+         }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->